### ISSUE TITLE
Use shiki-twoslash for .svelte code blocks

### ISSUE THREAD

**Issue by Rich-Harris** [sveltejs/svelte.dev#649]
### Describe the problem

We just upgraded the docs to use shiki-twoslash, which provides tooltips on `js` code blocks:

<img width="730" alt="image" src="https://user-images.githubusercontent.com/1162160/154771534-879b1fcd-1ff3-4baf-bd82-15d0db345965.png">

It would be cool if `svelte` blocks got the same treatment.

### Describe the proposed solution

Honestly no idea, but @jasonlyu123 got a POC working previously: https://github.com/sveltejs/kit/issues/3324#issuecomment-1024850076

### Importance

nice to have

**Comment by jasonlyu123** [comment]
Right, I never followed up on this. At some point, the POC became overly complicated when new stuff was added to the kit docs. The biggest issue is supporting the `// ---cut---` comment as the script tag needs to be partly cut so source-map is harder. There is also some ts diagnostics error in `svelte2tsx` generated code that needs to be filtered out in a weird way and also requires a different sourcemap. So there are like a total of 3 or 4 kinds of sourcemap.

**Comment by eltigerchino** [comment]
I assumed this was working when I saw the hover tooltips working on https://svelte.dev/docs/svelte/$state

**Comment by jasonlyu123** [comment]
Yeah. the library only works with ts/js so there isn't hover info in the svelte file. There is now a twoslash vue and eslint so maybe there would be more stuff added to the base library making it easier and not needing so many layers of source map.

**Comment by Hugos68** [comment]
I've made some efforts to build `twoslash/svelte`:
- PR: https://github.com/twoslashes/twoslash/pull/57
- Preview: https://deploy-preview-57--twoslash.netlify.app/packages/svelte

If this is something that you think will be valueable to the docs I will gladly submit a PR.
Just note that the PR is still just a PR so the package isn't out yet.

**Comment by huntabyte** [comment]
@Hugos68 I have been patiently waiting for someone to build a Svelte integration for this for quite some time. This is incredible and will benefit not only the svelte.dev site but the entire ecosystem of Svelte libraries 💪 incredible job!

**Comment by Rich-Harris** [comment]
> If this is something that you think will be valueable to the docs I will gladly submit a PR to add twoslash to `.svelte` blocks

This would be incredible, thank you!

**Comment by Hugos68** [comment]
Sure thing, once `twoslash-svelte` gets merged into `twoslash` I'll submit a PR 👍
